sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -769,10 +779,38 @@ void CAFMaker::beginSubRun(art::SubRun& sr) {
   if(bnb_spill){
     FillExposure(*bnb_spill, fBNBInfo, fSubRunPOT);
     fTotalPOT += fSubRunPOT;
+
+    // Find the spill for each event and fill the event map:
+    // We take the latest spill for a given event number to be the one to keep
+    fHasBNBInfo = true;
+    for(const sbn::BNBSpillInfo& info: *bnb_spill)
+    {
+      if ( fBNBInfoEventMap.find(info.event)==fBNBInfoEventMap.end() ) {
+        fBNBInfoEventMap[info.event] = info;
+      }
+      else if ( (info.spill_time_s+(info.spill_time_ns/1.0e9)) >
+                (fBNBInfoEventMap[info.event].spill_time_s+(fBNBInfoEventMap[info.event].spill_time_ns/1.0e9)) ) {
+        fBNBInfoEventMap[info.event] = info;
+      }
```

Comment:
Minor point: optimisation to reduce lookups in the map:
```suggestion
      auto const iInfo = fBNBInfoEventMap.find(info.event);
      if ( iInfo==fBNBInfoEventMap.end() ) {
        fBNBInfoEventMap[info.event] = info;
      }
      else if ( (info.spill_time_s+(info.spill_time_ns/1.0e9)) >
                (iInfo->spill_time_s+(iInfo->spill_time_ns/1.0e9)) ) {
        *iInfo = info;
      }
```
Another approach is a bit more convoluted:
```suggestion
      auto& storedInfo = fBNBInfoEventMap[info.event];
      if (storedInfo.event == UINT_MAX) { // no info
        storedInfo = info;
      }
      else if ( (info.spill_time_s+info.spill_time_ns/1.0e9) >
                (storedInfo.spill_time_s+storedInfo.spill_time_ns/1.0e9)) ) {
        storedInfo = info;
      }
```
which relies on the fact that a default-constructed `SpillInfo` object has an event number `UINT_MAX`; or, merging with the other suggestion:
```suggestion
      auto& storedInfo = fBNBInfoEventMap[info.event]; // creates if needed
      if ((storedInfo.event == UINT_MAX) // no info
        || spillInfoToTimestamp(info) > spillInfoToTimestamp(storedInfo) // newer
      {
        storedInfo = std::move(info);
      }
```